packages/template-retail-react-app/app/components/item-variant/item-price.jsx

```diff
@@ -43,74 +36,68 @@ PricePerItem.propTypes = {
  */
 const ItemPrice = ({currency, align = 'right', baseDirection = 'column', ...props}) => {
     const variant = useItemVariant()
-    const {data: basket} = useCurrentBasket()
-    const {currency: activeCurrency} = useCurrency()
-    const intl = useIntl()
-
-    const {price, basePrice, priceAfterItemDiscount} = variant
-    const isProductASet = variant?.type?.set
-
-    const displayPrice =
-        typeof priceAfterItemDiscount === 'number' ? Math.min(price, priceAfterItemDiscount) : price
-
-    const hasDiscount = displayPrice !== price
-
+    const {price, priceAfterItemDiscount} = variant
+    const isASet = variant?.type?.set
+    const isMaster = variant?.type?.master
+    let priceData
+    // this indicates the variant is on cart page
+    // when on cart page, the variant data is merged with basket item info,
+    // it will have price info that takes quantity and different promotions into the account
+    // we should prioritize use these values for display price
+    // and should NOT attempt to re-calculate at all cost here
+    if (variant?.itemId) {
+        priceData = {
+            currentPrice: priceAfterItemDiscount,
+            listPrice: price,
+            pricePerUnit: variant?.pricePerUnit,
+            isASet,
+            isMaster,
+            isRange: isASet || isMaster || false,
+            isOnSale: price > priceAfterItemDiscount
+        }
+    } else {
+        // for wishlist page we extract price info from variant/product obj
+        priceData = getPriceData(variant)
+    }
+    const [isDesktop] = useMediaQuery('(min-width: 992px)')
```

Comment:
I think I've written similar code, but this wouldn't fare well if someone wanted to customize/override the base sizes. Can you think of anything we can do here to make this code safer when there is changes to these width values on a application level.

Reply:
Good call, it is fixed